File: problem-solving/beecrowd/1061-tempo-evento/main.c
#include <stdio.h>

int main(){
  char  _[5];
  int starting_d, finishing_d, starting_h, finishing_h, starting_m, finishing_m, starting_s, finishing_s;
  int days, hours, minutes, seconds;

  scanf("%s %d", _, &starting_d);
  scanf("%d %s %d %s %d", &starting_h, _, &starting_m, _, &starting_s);
  scanf("%s %d", _, &finishing_d);
  scanf("%d %s %d %s %d", &finishing_h, _, &finishing_m, _, &finishing_s);

  days = finishing_d - starting_d;
  hours = finishing_h - starting_h;
  minutes = finishing_m - starting_m;
  seconds = finishing_s - starting_s;

  if (seconds < 0){
    seconds = seconds + 60;
    minutes--;
  }
  if (minutes < 0){
    minutes = minutes + 60;
    hours--;
  }
  if (hours < 0){
    hours = hours + 24;
    days--;
  }

  printf("%d dia(s)\n", days);
  printf("%d hora(s)\n", hours);
  printf("%d minuto(s)\n", minutes);
  printf("%d segundo(s)\n", seconds);
}

File: problem-solving/beecrowd/1061-tempo-evento/main.py
_, starting_d = input().split()
starting_h, starting_m,  starting_s = input().split(" : ")
_, finishing_d = input().split()
finishing_h, finishing_m, finishing_s = input().split(" : ")

starting_d, starting_h, starting_m, starting_s = int(starting_d), int(starting_h), int(starting_m), int(starting_s)
finishing_d, finishing_h, finishing_m, finishing_s = int(finishing_d), int(finishing_h), int(finishing_m), int(finishing_s)

days = finishing_d - starting_d
hours = finishing_h - starting_h
minutes = finishing_m - starting_m
seconds = finishing_s - starting_s

if (seconds < 0):
    seconds += 60
    minutes -=1

if (minutes < 0):
    minutes += 60
    hours -= 1

if (hours < 0):
    hours += 24
    days -= 1

print(
        f"{days} dia(s)",
        f"{hours} hora(s)",
        f"{minutes} minuto(s)",
        f"{seconds} segundo(s)",
        sep="\n"
       )
